python/cdp/network_scoped_evm_server_account.py

```diff
@@ -0,0 +1,377 @@
+            network=self._network,
+        )
+
+    async def _network_scoped_wait_for_transaction_receipt(
+        self,
+        transaction_hash: str,
+        timeout_seconds: float = 20,
+        interval_seconds: float = 0.2,
+        rpc_url: str | None = None,
+    ) -> dict:
+        """Wait for transaction receipt using web3.py if custom RPC is set, otherwise use the API."""
+        # If we have a custom RPC URL, use web3.py
+        if self._rpc_url:
+            if not self._web3:
+                # Initialize web3 if not already done
+                self._web3 = Web3(Web3.HTTPProvider(self._rpc_url))
+
+            import asyncio
+            from time import time
+
+            from web3.exceptions import TransactionNotFound
+
+            start = time()
+            while True:
+                try:
+                    receipt = self._web3.eth.get_transaction_receipt(transaction_hash)
+                    if receipt:
+                        return dict(receipt)
+                except TransactionNotFound:
+                    # Transaction not found yet, continue waiting
+                    pass
+
+                if time() - start > timeout_seconds:
+                    raise TimeoutError("Timeout waiting for transaction receipt via custom RPC.")
+                await asyncio.sleep(interval_seconds)
+        else:
+            # For managed networks, use default RPC URLs
+            # Since EvmServerAccount doesn't have a wait_for_transaction_receipt method
+            # We'll use web3.py directly with the network's default RPC
+            default_rpc_urls = {
+                "base": "https://mainnet.base.org",
+                "base-sepolia": "https://sepolia.base.org",
+                "ethereum": "https://eth.llamarpc.com",
+                "ethereum-sepolia": "https://rpc.sepolia.org",
+            }
+
+            network_rpc_url = default_rpc_urls.get(self._network)
+            if not network_rpc_url:
+                raise NotImplementedError(
+                    f"wait_for_transaction_receipt is not supported for network '{self._network}'. "
+                    "Please use web3.py directly or provide a custom RPC URL."
+                )
+
+            web3 = Web3(Web3.HTTPProvider(network_rpc_url))
+
+            import asyncio
+            from time import time
+
+            from web3.exceptions import TransactionNotFound
+
+            start = time()
+            while True:
+                try:
+                    receipt = web3.eth.get_transaction_receipt(transaction_hash)
+                    if receipt:
+                        return dict(receipt)
+                except TransactionNotFound:
+                    # Transaction not found yet, continue waiting
+                    pass
+
+                if time() - start > timeout_seconds:
+                    raise TimeoutError("Timeout waiting for transaction receipt.")
+                await asyncio.sleep(interval_seconds)
```

Comment:
this can be removed in favor of `web3.eth.wait_for_transaction_receipt`: https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.wait_for_transaction_receipt

Here's an example from our repo that uses it: https://github.com/coinbase/cdp-sdk/blob/main/examples/python/evm/account.send_transaction.py#L23

Reply:
(Tracking in a followup task)